packages/cli/src/codegen.ts

```diff
@@ -405,8 +405,109 @@ function genContracts(outDir: string, artifactDir: string, exports: string[]) {
+// and the second array contains contracts not deployed on all networks.
+function getFields(arrays: string[][]): [string[], string[]] {
+  if (arrays.length === 0) {
+    return [[], []]
+  }
+  const firstArray = arrays[0]
+  if (arrays.length === 1) {
+    return [firstArray, []]
+  }
+  const remains = arrays.slice(1)
+  const fields = firstArray.filter((elem) => remains.every((arr) => arr.includes(elem)))
+  const allElements = arrays.reduce((a, b) => a.concat(b)).filter((elem, index, arr) => arr.indexOf(elem) === index)
+  const optionalFields = allElements.filter((elem) => !fields.includes(elem))
+  return [fields, optionalFields]
+}
+
+async function genDeploymentsType(config: Configuration): Promise<string> {
+  const filePaths = networkIds
+    .map((n) => getDeploymentFilePath(config, n))
+    .filter((filePath) => fs.existsSync(filePath))
+  const allDeployments: DeploymentsPerAddress[] = []
+  for (const filePath of filePaths) {
+    const deployments = await Deployments.from(filePath)
+    if (deployments.deployments.length > 0) {
+      allDeployments.push(deployments.deployments[0])
+    }
+  }
+  const allContracts = allDeployments.map((d) => Array.from(d.contracts.keys()))
+  const allScripts = allDeployments.map((d) => Array.from(d.scripts.keys()))
+  const [contracts, optionalContracts] = getFields(allContracts)
+  const [scripts, optionalScripts] = getFields(allScripts)
+  const contractFields = contracts.map((name) => `${name}: DeployContractExecutionResult`).join('\n')
+  const optionalContractFields = optionalContracts.map((name) => `${name}?: DeployContractExecutionResult`).join('\n')
+  const scriptFields = scripts.map((name) => `${name}: RunScriptResult`).join('\n')
+  const optionalScriptFields = optionalScripts.map((name) => `${name}?: RunScriptResult`).join('\n')
+  const hasScript = scripts.length > 0 || optionalScripts.length > 0
+  return `
+    export type Deployments = {
+      deployerAddress: string
+      contracts: { ${contractFields} ${optionalContractFields} }
+      ${hasScript ? `scripts: { ${scriptFields} ${optionalScriptFields} }` : ''}
+    }
+  `
+}
+
+function getRelativePath(config: Configuration, networkId: NetworkId, fromPath: string): string | undefined {
+  const deploymentFilePath = getDeploymentFilePath(config, networkId)
+  if (!fs.existsSync(path.resolve(deploymentFilePath))) {
+    return undefined
+  }
+  return path.relative(fromPath, deploymentFilePath)
+}
+
+export async function genLoadDeployments(config: Configuration) {
+  const tsPath = path.join(config.artifactDir ?? DEFAULT_CONFIGURATION_VALUES.artifactDir, 'ts')
+  const deploymentsPath = networkIds.map((n) => [n, getRelativePath(config, n, tsPath)])
+  const deploymentsExists = deploymentsPath.filter(([, filePath]) => filePath !== undefined)
+  const imports = deploymentsExists
+    .map(([networkId, relativePath]) => {
+      return `import { default as ${networkId}Deployments } from '${relativePath}'`
```

Comment:
We need to import the JSON files because the front-end cannot dynamically read files based on paths.